src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.h

```diff
@@ -0,0 +1,199 @@
+#pragma once
+
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/hl/stp/tactic/transition_conditions.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/geom/algorithms/acute_angle.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/intersection.h"
+#include "software/geom/ray.h"
+#include "software/logger/logger.h"
+
+// Which way the crease defender should be aligned
+MAKE_ENUM(CreaseDefenderAlignment, LEFT, RIGHT, CENTRE);
+
+struct CreaseDefenderFSM
+{
+    class BlockThreatState;
+
+    // this struct defines the unique control parameters that the CreaseDefenderFSM
+    // requires in its update
+    struct ControlParams
+    {
+        Point enemy_threat_origin;
+        CreaseDefenderAlignment crease_defender_alignment;
+        double robot_obstacle_inflation_factor;
+    };
+
+    // this struct defines the only event that the CreaseDefenderFSM responds to
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    /**
+     * Finds the point to block the threat
+     *
+     * @param field The field
+     * @param enemy_threat_origin The origin of the threat to defend against
+     * @param crease_defender_alignment alignment of the crease defender
+     * @param robot_obstacle_inflation_factor The robot obstacle inflation factor
+     *
+     * @return The best point to block the threat if it exists
+     */
+    static std::optional<Point> findBlockThreatPoint(
+        const Field& field, const Point& enemy_threat_origin,
+        const CreaseDefenderAlignment& crease_defender_alignment,
+        double robot_obstacle_inflation_factor)
+    {
+        Angle shot_angle_increment =
+            acuteAngle(field.friendlyGoalpostPos(), enemy_threat_origin,
+                       field.friendlyGoalpostNeg()) /
+            6.0;
```

Comment:
what's the purpose of incrementing the angle?

Reply:
We increment either by 1*shot_angle_increment, 3*shot_angle_increment, or 5*shot_angle_increment. I cleaned up the comments and variable naming on that